llvm/include/llvm/CodeGen/GlobalISel/LegalizationArtifactCombiner.h

```diff
@@ -99,6 +100,8 @@ class LegalizationArtifactCombiner {
       const LLT DstTy = MRI.getType(DstReg);
       if (isInstLegal({TargetOpcode::G_CONSTANT, {DstTy}})) {
         auto &CstVal = SrcMI->getOperand(1);
+        Builder.MergedLocation = DILocation::getMergedLocation(
```

Comment:
I don't think I like this solution of keeping global state in the MachineBuilder over threading it through the API.

Reply:
@arsenm I concur with Adrian, one of the issues with the current approach is that there are multiple code paths that reach `CSEMIRBuilder::getDominatingInstrForID` so I have to clear the `MergedLocation` right after using it for the dominating instruction. If pass the DILocation through the API, we can just have a default nullptr value on it, and it doesn't change much.